src/draw/projection.typ

```diff
@@ -0,0 +1,115 @@
+#import "grouping.typ": group, get-ctx
+#import "transformations.typ": set-transform
+#import "/src/process.typ"
+#import "/src/matrix.typ"
+#import "/src/drawable.typ"
+#import "/src/util.typ"
+
+#let ortho-projection-matrix = ((1, 0, 0, 0),
+                                (0, 1, 0, 0),
+                                (0, 0, 0, 0),
+                                (0, 0, 0, 1))
+
+#let ortho-matrix(x, y, z) = matrix.mul-mat(
+  matrix.mul-mat(
+    matrix.mul-mat(
+      matrix.ident(),
+      matrix.transform-rotate-x(x)
+    ),
+    matrix.transform-rotate-y(y)
+  ),
+  matrix.transform-rotate-z(z)
+)
+
+
+#let _projection(body, view-matrix, projection-matrix, reset-transform: false) = {
+  (ctx => {
+    let transform = ctx.transform
+    ctx.transform = matrix.mul-mat(projection-matrix, view-matrix)
+    if not reset-transform {
+      ctx.transform = matrix.mul-mat(transform, ctx.transform)
+    }
+    let (ctx, drawables, bounds) = process.many(ctx, body)
+    ctx.transform = transform
+
+    return (
+      ctx: ctx,
+      bounds: bounds,
+      drawables: drawables,
+    )
+  },)
+}
+
+#let apply-transform-function(body, fn) = {
+  (ctx => {
+    let transform = ctx.transform
+    ctx.transform = matrix.ident()
+    let (ctx, drawables, bounds) = process.many(ctx, body)
+    ctx.transform = transform
+
+    drawables = drawables.map(d => {
+      d.segments = d.segments.map(s => {
+        let (kind, ..pts) = s
+        return (kind, ..pts.map(pt => {
+          fn(pt)
+        }))
+      })
+      if d.type == "content" {
+        d.pos = fn(d.pos)
+      }
+      return d
+    })
```

Comment:
Is this really the best way to do this? Is there no transform that could be used instead?

Could you clear up the segments map as well maybe:
```
(s.slice(0, 1), ..s.slice(1).map(...))
```

Reply:
What do you mean by "clear up"?
You should be able to get the same result using rotation + scaling + translation.

Reply:
It looks difficult to read, so tidy it up so it reads better:
```
d.segments.map(s => { (s.slice(0, 1), ..s.slice(1).map(fn)) })
```